src/Collection.php

```diff
@@ -154,14 +153,14 @@ final class Collection implements CollectionInterface
     private array $parameters;
 
     /**
-     * @var callable(mixed ...$parameters): iterable<TKey, T>
+     * @var callable(mixed ...$parameters): Iterator<TKey, T>
```

Comment:
this change means we no longer want to allow this?

```PHP
Collection::fromCallable(static fn(): array => [1, 2, 3])
```

I don't see why this would not be allowed because `fromIterable` allows `array`, i.e.:
```PHP
Collection::fromIterable([1, 2, 3])
```

Reply:
I guess we can argue that we want to keep it to only one type `Iterator`, but it feels like a restriction which should be justified in some way in the documentation

Reply:
This is now no more valid, I got a better idea, I let you check it out ;)